okhttp/src/main/java/okhttp3/internal/connection/StreamAllocation.java

```diff
@@ -106,10 +106,16 @@ public HttpCodec newStream(
     boolean connectionRetryEnabled = client.retryOnConnectionFailure();
 
     try {
+      RealConnection existingConnection = connection;
+
       RealConnection resultConnection = findHealthyConnection(connectTimeout, readTimeout,
           writeTimeout, connectionRetryEnabled, doExtensiveHealthChecks);
       HttpCodec resultCodec = resultConnection.newCodec(client, chain, this);
 
+      if (existingConnection != connection) {
```

Comment:
That was intentional. It will get called for fresh and recycled connections. It was suggested by @swankjesse unless I misunderstood.